### PR TITLE
Sonar: Immediately return this expression instead of assigning it to …

### PR DESCRIPTION
**Pull request by slawekk87**
…the temporary variable

## Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ ] My code is written according to the guidelines found [here] (https://yetiforce.com/en/knowledge-base/documentation/developer-documentation/item/how-to-create-php-files).
- [ ] I have signed the Contributor Licence Agreement between me and YetiForce.

### CODE REVIEW

File: modules/Settings/CurrencyUpdate/models/Module.php [hunk 1]
```diff
@@ -186,9 +186,7 @@ public function getRatesHistory($bankId, $dateCur, \App\Request $request)
 		}
 		$query->orderBy(['exchange_date' => SORT_DESC, 'currency_code' => SORT_ASC]);
 		$dataReader = $query->createCommand()->query();
-		$history = $dataReader->readAll();
-
-		return $history;
+		return $dataReader->readAll();
```
Comment (TomaszQr) [inline modules/Settings/CurrencyUpdate/models/Module.php]: The variable $dataReader is used only once

File: modules/Settings/LangManagement/views/Index.php [hunk 2]
```diff
@@ -35,19 +35,16 @@ public function process(\App\Request $request)
 	 */
 	public function getFooterScripts(\App\Request $request)
 	{
-		$moduleName = $request->getModule();
-		$jsFileNames = [
-			"modules.Settings.$moduleName.resources.LangManagement",
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			'modules.Settings.' . $request->getModule() . '.resources.LangManagement',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/LangManagement/views/Index.php]: "modules.Settings.{$request->getModule()}.resources.LangManagement"

File: modules/Settings/Leads/views/MappingDetail.php [hunk 3]
```diff
@@ -30,16 +30,8 @@ public function process(\App\Request $request)
...
-		$jsFileNames = [
-			"modules.Settings.$moduleName.resources.LeadMapping",
-		];
-
-		$jsScriptInstances = $this->checkAndConvertJsScripts($jsFileNames);
-		$headerScriptInstances = array_merge($headerScriptInstances, $jsScriptInstances);
-
-		return $headerScriptInstances;
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			'modules.Settings.' . $request->getModule() . '.resources.LeadMapping',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Leads/views/MappingDetail.php]: "modules.Settings.{$request->getModule()}.resources.LeadMapping"

File: modules/Settings/Leads/views/MappingEdit.php [hunk 4]
```diff
@@ -33,16 +33,8 @@ public function process(\App\Request $request)
...
-		$jsFileNames = [
-			"modules.Settings.$moduleName.resources.LeadMapping",
-		];
-
-		$jsScriptInstances = $this->checkAndConvertJsScripts($jsFileNames);
-		$headerScriptInstances = array_merge($headerScriptInstances, $jsScriptInstances);
-
-		return $headerScriptInstances;
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			'modules.Settings.' . $request->getModule() . '.resources.LeadMapping',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Leads/views/MappingEdit.php]: ?

File: modules/Settings/Mail/views/Autologin.php [hunk 5]
```diff
@@ -33,16 +33,8 @@ public function process(\App\Request $request)
...
-		$jsFileNames = [
-			"modules.Settings.$moduleName.resources.Autologin",
-		];
-
-		$jsScriptInstances = $this->checkAndConvertJsScripts($jsFileNames);
-		$headerScriptInstances = array_merge($headerScriptInstances, $jsScriptInstances);
-
-		return $headerScriptInstances;
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			'modules.Settings.' . $request->getModule() . '.resources.Autologin',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Mail/views/Autologin.php]: ?

File: modules/Settings/Mail/views/Config.php [hunk 6]
```diff
@@ -33,17 +33,9 @@ public function process(\App\Request $request)
 	 */
 	public function getFooterScripts(\App\Request $request)
 	{
-		$headerScriptInstances = parent::getFooterScripts($request);
-		$moduleName = $request->getModule();
-
-		$jsFileNames = [
-			"modules.Settings.$moduleName.resources.Config",
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			'modules.Settings.' . $request->getModule() . '.resources.Config',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Mail/views/Config.php]: ?

File: modules/Settings/Mail/views/Detail.php [hunk 7]
```diff
@@ -60,16 +60,9 @@ public function process(\App\Request $request)
...
-		$jsFileNames = [
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
 			'modules.Settings.Vtiger.resources.Detail',
-			"modules.Settings.$moduleName.resources.Detail",
-		];
-		$jsScriptInstances = $this->checkAndConvertJsScripts($jsFileNames);
-		$headerScriptInstances = array_merge($headerScriptInstances, $jsScriptInstances);
-
-		return $headerScriptInstances;
+			'modules.Settings.' . $request->getModule() . '.resources.Detail',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Mail/views/Detail.php]: ?

File: modules/Settings/Menu/views/Index.php [hunk 8]
```diff
@@ -36,28 +36,16 @@ public function process(\App\Request $request)
...
-		$cssFileNames = [
+		return array_merge($this->checkAndConvertCssStyles([
 			'~libraries/jstree/dist/themes/default/style.css',
-			"modules.Settings.$moduleName.Index",
-		];
-		$cssInstances = $this->checkAndConvertCssStyles($cssFileNames);
-		$headerCssInstances = array_merge($cssInstances, $headerCssInstances);
-
-		return $headerCssInstances;
+			'modules.Settings.' . $request->getModule() . '.Index',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Menu/views/Index.php]: ?

File: modules/Settings/ModuleManager/views/ModuleImport.php [hunk 9]
```diff
@@ -51,17 +51,9 @@ public function process(\App\Request $request)
...
-		$jsFileNames = [
-			"modules.Settings.$moduleName.resources.ModuleImport",
-		];
-
-		$jsScriptInstances = $this->checkAndConvertJsScripts($jsFileNames);
-		$headerScriptInstances = array_merge($headerScriptInstances, $jsScriptInstances);
-
-		return $headerScriptInstances;
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			"modules.Settings.'.$request->getModule().'.resources.ModuleImport",
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/ModuleManager/views/ModuleImport.php]: ?

File: modules/Settings/Notifications/views/Configuration.php [hunk 10]
```diff
@@ -39,18 +39,12 @@ public function process(\App\Request $request)
 	 */
 	public function getFooterScripts(\App\Request $request)
 	{
-		$headerScriptInstances = parent::getFooterScripts($request);
-		$moduleName = $request->getModule();
-		$jsFileNames = [
-			"modules.Settings.$moduleName.resources.Configuration",
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			'modules.Settings.' . $request->getModule() . '.resources.Configuration',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Notifications/views/Configuration.php]: ?

File: modules/Settings/PDF/views/Edit.php [hunk 11]
```diff
@@ -79,24 +77,13 @@ public function getFooterScripts(\App\Request $request)
...
-		$moduleName = $request->getModule();
-		$cssFileNames = [
-			"modules.Settings.$moduleName.Edit",
-		];
-		$cssInstances = $this->checkAndConvertCssStyles($cssFileNames);
-		$headerCssInstances = array_merge($cssInstances, $headerCssInstances);
-
-		return $headerCssInstances;
+		return array_merge($this->checkAndConvertCssStyles([
+			'modules.Settings.' . $request->getModule() . '.Edit',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/PDF/views/Edit.php]: ?

File: modules/Settings/PDF/views/Import.php [hunk 12]
```diff
@@ -73,14 +73,8 @@ public function process(\App\Request $request)
...
-		$moduleName = $request->getModule();
-		$cssFileNames = [
-			"modules.Settings.$moduleName.Edit",
-		];
-		$cssInstances = $this->checkAndConvertCssStyles($cssFileNames);
-		$headerCssInstances = array_merge($cssInstances, $headerCssInstances);
-
-		return $headerCssInstances;
+		return array_merge($this->checkAndConvertCssStyles([
+			'modules.Settings.' . $request->getModule() . '.Edit',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/PDF/views/Import.php]: ?

File: modules/Settings/Password/views/Index.php [hunk 13]
```diff
@@ -20,16 +20,8 @@ public function process(\App\Request $request)
...
-		$jsFileNames = [
-			"modules.Settings.$moduleName.resources.Password",
-		];
-
-		$jsScriptInstances = $this->checkAndConvertJsScripts($jsFileNames);
-		$headerScriptInstances = array_merge($headerScriptInstances, $jsScriptInstances);
-
-		return $headerScriptInstances;
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			'modules.Settings.' . $request->getModule() . '.resources.Password',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Password/views/Index.php]: ?

File: modules/Settings/Profiles/views/Edit.php [hunk 14]
```diff
@@ -72,17 +72,9 @@ public function initialize(\App\Request $request)
...
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
 			'modules.Settings.Vtiger.resources.Edit',
-			"modules.Settings.$moduleName.resources.Edit",
-		];
-
-		$jsScriptInstances = $this->checkAndConvertJsScripts($jsFileNames);
-		$headerScriptInstances = array_merge($headerScriptInstances, $jsScriptInstances);
-
-		return $headerScriptInstances;
+			'modules.Settings.' . $request->getModule() . '.resources.Edit',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Profiles/views/Edit.php]: ?

File: modules/Settings/SalesProcesses/views/Index.php [hunk 15]
```diff
@@ -28,16 +28,8 @@ public function process(\App\Request $request)
...
-		$jsFileNames = [
-			"modules.Settings.$moduleName.resources.Index",
-		];
-
-		$jsScriptInstances = $this->checkAndConvertJsScripts($jsFileNames);
-		$headerScriptInstances = array_merge($headerScriptInstances, $jsScriptInstances);
-
-		return $headerScriptInstances;
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			'modules.Settings.' . $request->getModule() . '.resources.Index',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/SalesProcesses/views/Index.php]: ?

**Pull request merged**